em_examples/HarmonicVMDCylWidget.py

```diff
@@ -181,139 +246,193 @@ def getData(self, bType="b"):
-                    val = self.Bz.imag
-                elif ComplexNumber == "amplitude":
-                    val = abs(self.Bz)
-                elif ComplexNumber == "phase":
-                    val = np.angle(self.Bz)
-            else:
-                # ax.imshow(self.im)
-                ax.set_xticks([])
-                ax.set_yticks([])
-                return "Dude, think twice ... no By for VMD"
-
-        elif Field == "E":
-            label = "Electric field (V/m)"
-            if view == "y":
-                if ComplexNumber == "real":
-                    val = self.Ey.real
-                elif ComplexNumber == "imag":
-                    val = self.Ey.imag
-                elif ComplexNumber == "amplitude":
-                    val = abs(self.Ey)
-                elif ComplexNumber == "phase":
-                    val = np.angle(self.Ey)
-            else:
-                # ax.imshow(self.im)
-                ax.set_xticks([])
-                ax.set_yticks([])
-                return "Dude, think twice ... only Ey for VMD"
-
-        elif Field == "J":
-            label = "Current density (A/m$^2$)"
-            if view == "y":
-                if ComplexNumber == "real":
-                    val = self.Jy.real
-                elif ComplexNumber == "imag":
-                    val = self.Jy.imag
-                elif ComplexNumber == "amplitude":
-                    val = abs(self.Jy)
-                elif ComplexNumber == "phase":
-                    val = np.angle(self.Jy)
-            else:
-                # ax.imshow(self.im)
-                ax.set_xticks([])
-                ax.set_yticks([])
-                return "Dude, think twice ... only Jy for VMD"
-
-        out = Utils.plot2Ddata(
-            self.mesh2D.gridCC, val, vec=vec, ax=ax,
-            contourOpts={"cmap": "viridis"}, ncontour=50, scale=scale
-        )
-        if scale == "linear":
-            cb = plt.colorbar(
-                out[0], ax=ax, ticks=np.linspace(val.min(), val.max(), 3),
-                format="%.1e"
-            )
-        elif scale == "log":
-            cb = plt.colorbar(
-                out[0], ax=ax, ticks=np.linspace(val.min(), val.max(), 3),
-                format="$10^{%.1f}$"
+            elif Field == "E":
+                label = "Electric field (V/m)"
+                if view == "y":
+                    if ComplexNumber == "real":
+                        val = self.Ey.real
+                    elif ComplexNumber == "imag":
+                        val = self.Ey.imag
+                    elif ComplexNumber == "amplitude":
+                        val = abs(self.Ey)
+                    elif ComplexNumber == "phase":
+                        val = np.angle(self.Ey)
+                else:
+                    # ax.imshow(self.im)
+                    ax.set_xticks([])
+                    ax.set_yticks([])
+                    print("Think about the problem geometry. There is NO Ex or Ez in this case.")
```

Comment:
same comment: it is also worth adding a `return` statement right after this (no need to draw the figure, printing is enough)